signature/README.md

```diff
@@ -0,0 +1,174 @@
+like this
+
+```rust
+#[derive(Deserialize, Debug, PartialEq, Serialize)]
+pub struct NewSignSchemeParameters {
+    #[serde(rename = "signature-path")]
+    pub signature_path: String,
+}
+```
+And then the field can be deserialized from `policy.json`.
+
+Besides, Implement the trait `SignScheme` for `NewSignSchemeParameters`.
+```rust
+pub trait SignScheme {
+    /// Prepare runtime directories for storing signatures, configuretions ,.etc
+    fn prepare_runtime_dirs(&self) -> Result<()>;
+
+    /// Check whether the required resources need to be obtained from KBS.
+    fn resources_check(&self) -> Result<bool>;
+
+    /// Get the resources list. The list contains resources names to be
+    /// got by the kbs, using `get_resource_from_kbs()` in validate.rs
+    fn needed_resources_list_from_kbs(&self) -> Vec<&str>;
+
+    /// Process all the gathered resources from kbs. All the resources
+    /// gathered from kbs due to the return value of `needed_resources_list_from_kbs()`
+    /// will be recorded into a HashMap, mapping `resource name` ->
+    /// `content of Vec<u8>`. The parameter of this function is the
+    /// HashMap.
+    fn process_gathered_resources(&self, resources: HashMap<&str, Vec<u8>>) -> Result<()>;
+
+    /// Judge whether an image is allowed by this SignScheme.
+    fn allows_image(&self, image: &mut Image) -> Result<()>;
+}
+```
+
+For a specific signing scheme in a Policy Requirement,
+these 5 functions are called in the following order.
+
+```plaintext
++------------+                                          +-----------+
+|            |         1.prepare_runtime_dirs()         |           |
+|            +----------------------------------------->|           |
+|            |                                          |           |
+|            |         2.resources_check()              |           |
+|            +----------------------------------------->|           |
+|            |                                          | signature |
+|  image-rs  |    3.needed_resources_list_from_kbs()    |  schemes  |
+|            +----------------------------------------->|           |
+|            |                                          |           |
+|            |         4.get_scheme_resources()         |           |
+|            +----------------------------------------->|           |
+|            |                                          |           |
+|            |        5.process_gathered_resources      |           |
+|            +----------------------------------------->|           |
+|            |                                          |           |
+|            |                                          |           |
+|            |                                          |           |
++------------+                                          +-----------+
+```
+
+* Firstly, `prepare_runtime_dirs()` will prepare the directories that this
```

Comment:
I wonder whether the five APIs can be more concise. For example, the second, third and fourth APIs are actually dealing with the resources required for verification, so can they be encapsulated in the same API?

Reply:
Good idea. I tried this, but the function `get_scheme_resources()` in [validate.rs](https://github.com/confidential-containers/image-rs/blob/main/src/validate.rs) is a async function, while all functions in the signature module are sync functions. The way to deal this is to refactor the signature module deeper to make relative functions async.

Reply:
Thank you for your advice. [I have made the related functions async](https://github.com/confidential-containers/image-rs/pull/39/commits/381460b00ad208600c3e255181fd419d818b467d). But also, for the 2nd reason, may not merge the two functions.